applications/RomApplication/python_scripts/calculate_rom_basis_output_process.py

```diff
@@ -0,0 +1,149 @@
+            "help": "A process to set the snapshots matrix and calculate the ROM basis from it.",
+            "model_part_name": "",
+            "snapshots_control_type": "step",
+            "snapshots_interval": 1.0,
+            "nodal_unknowns": [],
+            "rom_basis_output_format": "json",
+            "rom_basis_output_name": "RomParameters",
+            "svd_truncation_tolerance": 1.0e-6
+        }""")
+
+        return default_settings
+
+    def IsOutputStep(self):
+        if self.snapshots_control_is_time:
+            time = self.__GetPrettyFloat(self.model_part.ProcessInfo[KratosMultiphysics.TIME])
+            return time >= self.__GetPrettyFloat(self.next_output)
+        else:
+            step = self.__GetPrettyFloat(self.model_part.ProcessInfo[KratosMultiphysics.STEP])
+            return step >= self.next_output
+
+    def PrintOutput(self):
+        # Save the data in the snapshots data list
+        aux_data_array = []
+        for node in self.model_part.Nodes:
+            for snapshot_var in self.snapshot_variables_list:
+                aux_data_array.append(node.GetSolutionStepValue(snapshot_var))
+        self.snapshots_data_list.append(aux_data_array)
+
+        # Schedule next snapshot output
+        if self.snapshots_interval > 0.0: # Note: if == 0, we'll just always print
+            if self.snapshots_control_is_time:
+                time = self.__GetPrettyFloat(self.model_part.ProcessInfo[KratosMultiphysics.TIME])
+                while self.__GetPrettyFloat(self.next_output) <= time:
+                    self.next_output += self.snapshots_interval
+            else:
+                step = self.__GetPrettyFloat(self.model_part.ProcessInfo[KratosMultiphysics.STEP])
+                while self.next_output <= step:
+                    self.next_output += self.snapshots_interval
+
+    def ExecuteFinalize(self):
+        # Set a NumPy array with the snapshots data
+        n_nodes = self.model_part.NumberOfNodes()
+        n_data_cols = len(self.snapshots_data_list)
+        n_nodal_unknowns = len(self.snapshot_variables_list)
+        snapshots_matrix = numpy.empty((n_nodal_unknowns*n_nodes,n_data_cols))
+        for i_col in range(n_data_cols):
+            aux_col = numpy.array(self.snapshots_data_list[i_col])
+            snapshots_matrix[:,i_col] = aux_col.transpose()
+
+        # Calculate the randomized SVD of the snapshots matrix
+        u,_,_,_= RandomizedSingularValueDecomposition().Calculate(snapshots_matrix, self.svd_truncation_tolerance)
+
+        # Save the nodal basis
+        rom_basis_dict = {
+            "rom_settings": {},
+            "nodal_modes": {}
+        }
+        rom_basis_dict["rom_settings"]["nodal_unknowns"] = [var.Name() for var in self.snapshot_variables_list]
+        rom_basis_dict["rom_settings"]["number_of_rom_dofs"] = numpy.shape(u)[1] #TODO: This is way misleading. I'd call it number_of_basis_modes or number_of_rom_modes
```

Comment:
I agree the name should be changed. This dates back from the very first implementation of the RomApplication, and the `number_of_basis_modes` is a better choice

Reply:
Cool. My suggestion would be to stick to ` number_of_rom_dofs` for a while and, once we finish upgrading the application, we do a single PR with a search & replace.